packages/vite/src/node/config.ts

```diff
@@ -566,6 +566,33 @@ export async function resolveConfig(
     )
   }
 
+  // Check if all assetFileNames functions have the same reference.
+  // If not, display a warn for user.
+  const outputOption = config.build?.rollupOptions?.output ?? []
+  // Use isArray to narrow its type to array
+  if (Array.isArray(outputOption)) {
+    const assetFileNamesFunctions = outputOption
+      .map((output) => output.assetFileNames)
+      .filter(
+        (assetFileNames) => typeof assetFileNames === 'function'
+      ) as Exclude<OutputOptions['assetFileNames'], undefined | string>[]
```

Comment:
We need to check this not only for functions, but also check any kind of option. This filter can be removed.